app/views/reports/regions/_child_data_tables_with_exclusions.html.erb

```diff
@@ -57,64 +57,61 @@
             <%= @region.name %>
           </td>
           <td class="ta-left">
-            <%= number_to_percentage(@data.last_value(:controlled_patients_rate), precision: 0) %>
+            <%= number_to_percentage(@data.dig(:controlled_patients_rate, @period), precision: 0) %>
           </td>
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:controlled_patients)) %>
+            <%= number_with_delimiter(@data.dig(:controlled_patients, @period)) %>
           </td>
           <td class="ta-left">
-            <%= number_to_percentage(@data.last_value(:uncontrolled_patients_rate), precision: 0) %>
+            <%= number_to_percentage(@data.dig(:uncontrolled_patients_rate, @period), precision: 0) %>
           </td>
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:uncontrolled_patients)) %>
+            <%= number_with_delimiter(@data.dig(:uncontrolled_patients, @period)) %>
           </td>
           <td class="ta-left">
-            <%= number_to_percentage(@data.last_value(:missed_visits_rate), precision: 0) %>
+            <%= number_to_percentage(@data.dig(:missed_visits_rate, @period), precision: 0) %>
           </td>
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:missed_visits)) %>
+            <%= number_with_delimiter(@data.dig(:missed_visits, @period)) %>
           </td>
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:cumulative_registrations)) %>
+            <%= number_with_delimiter(@data.dig(:cumulative_registrations, @period)) %>
           </td>
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:registrations)) %>
+            <%= number_with_delimiter(@data.dig(:registrations, @period)) %>
           </td>
         </tr>
         <% data.each do |result| %>
-          <% next if result.dig("controlled_patients_rate", @period).nil? %>
```

Comment:
I removed these (and the `|| 0` code below) because all of these except missed_visits_rate are all hashes that are instantiated to return nil instead of zero. Frankly, I'm not even sure why we want to skip to the next result if this is nil. That seems wrong to me. Does anyone know why that is?

Reply:
There are cases where the data for a facility can be completely empty for new facilities. With the positional approach, we would hit errors because values would be completely empty, so values.last would blow up. 

With the direct dig approach, I think we will rule out those types of errors altogether. 👍🏼